command-expansion-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -2,46 +2,136 @@
 export class Command<A extends ArgType[] | { [argName: string]: any } = [] | {}> {
   public readonly stem: string;
   public readonly arguments: A;
+  public readonly absoluteTime: Temporal.Instant | null;
+  public readonly epochTime: { epochName: string; time: Temporal.Duration } | null;
+  public readonly relativeTime: Temporal.Duration | null;
 
-  private constructor(opts: { stem: string; arguments: A }) {
+  private constructor(opts: { stem: string; arguments: A } & timingType) {
     this.stem = opts.stem;
     this.arguments = opts.arguments;
+    if ('absoluteTime' in opts) {
+      this.absoluteTime = opts.absoluteTime;
+      this.epochTime = null;
```

Comment:
100% a stylistic preference: We can have the properties initialize to `null` by doing like `public readonly absoluteTime: Temporal.Instant | null = null;` and then we don't have to assign here.

Reply:
I agree, less code in the file too :)